http-server/src/test/java/io/airlift/http/server/TestHttpServerCipher.java

```diff
@@ -216,19 +217,25 @@ private static HttpServer createServer(NodeInfo nodeInfo, HttpServerInfo httpSer
 
     private static HttpServer createServer(HttpServlet servlet, NodeInfo nodeInfo, HttpServerInfo httpServerInfo, HttpServerConfig config, HttpsConfig httpsConfig)
     {
-        return new HttpServerProvider(
-                httpServerInfo,
-                nodeInfo,
-                config,
-                Optional.of(httpsConfig),
-                servlet,
-                ImmutableSet.of(new DummyFilter()),
-                ImmutableSet.of(),
-                false,
-                false,
-                false,
-                ClientCertificate.NONE,
-                Optional.empty(),
-                Optional.empty()).get();
+        try {
+            HttpServer httpServer = new HttpServer(
+                    httpServerInfo,
+                    nodeInfo,
+                    config,
+                    Optional.of(httpsConfig),
+                    servlet,
+                    ImmutableSet.of(new DummyFilter()),
+                    ImmutableSet.of(),
+                    ServerFeature.defaults(),
+                    ClientCertificate.NONE,
+                    new TestingMBeanServer(),
+                    Optional.empty(),
+                    Optional.empty());
+            httpServer.start();
+            return httpServer;
+        }
+        catch (Exception e) {
```

Comment:
Don't catch `Exception` as this can mask `InterruptedException`. Catch the checked exceptions explicitly.
```java
catch (IOException e) {
    throw new UncheckedIOException(e);
}
```
Same for the others below